### ISSUE TITLE
Fails to update due to missing key

### ISSUE THREAD

**Issue by tomwmth** [kpcyrd/spotify-launcher#67]
Seems related to #41 - I assume the key changed again even though we aren't quite in January this time

```
[2024-12-27T13:48:08Z INFO  spotify_launcher::config] Loading configuration file at "/etc/spotify-launcher.conf"
[2024-12-27T13:48:08Z INFO  spotify_launcher::apt] Downloading release file...
[2024-12-27T13:48:09Z INFO  spotify_launcher::apt] Downloading signature...
[2024-12-27T13:48:10Z INFO  spotify_launcher::apt] Verifying pgp signature...
Missing key B420FD3777CCE3A7F0076B55C85668DF69375001, which is needed to verify signature.
Error: Verification of pgp signature didn't succeed
```

Running with `--skip-update` works as a temporary workaround

**Comment by j-palms** [comment]
Until this gets fixed you can apply the new key like this (run as root):

```
wget https://download.spotify.com/debian/pubkey_C85668DF69375001.gpg \
      -O /usr/share/spotify-launcher/keyring.pgp
```
(adjusted the solution from last year by @kpcyrd in https://github.com/kpcyrd/spotify-launcher/issues/41#issuecomment-1908001154)

**Comment by Seres67** [comment]
How did you find the url? I tried to look but couldn't find anything.

**Comment by j-palms** [comment]
I figured out that key id in the url is simply the last 16 chars of the full fingerprint ;)

**Comment by organizedgrime** [comment]
wouldn't it be nicer if the program automatically searched some keyserver for the missing key and added it to the keyring? it may be infrequent, but there will always be new keys to add to the keyring

**Comment by Mirukuu** [comment]
spotify-launcher --skip-update works too

**Comment by kpcyrd** [comment]
oh noes /o\

**Comment by kpcyrd** [comment]
This has been fixed in `spotify-launcher-0.6.1-1-x86_64.pkg.tar.zst` which should be available on an Arch Linux mirror near you very soon™.

Manjaro users please ping your distro to cherry-pick this update, there's nothing I can do there.

Sorry for failing you all with not making this kind of error non-fatal since the last time. </3

> wouldn't it be nicer if the program automatically searched some keyserver for the missing key and added it to the keyring? it may be infrequent, but there will always be new keys to add to the keyring

The key is pinned for a reason, implementing this effectively means "accept any signature from any key", which defeats the purpose of update signing. Trust needs to be assigned explicitly.

Happy 38c3 to all those who celebrate, see you at the Arch Linux user meetup tomorrow. :sparkles:

**Comment by kpcyrd** [comment]
I've pinged manjaro myself: https://forum.manjaro.org/t/update-requests-for-packages-imported-from-arch-linux/493/381?u=kpcyrd
